Linh - Implement Bluesky auto-poster backend

linh2020 opened the pull request:
# Description
This PR implements the backend for Bluesky auto-posting functionality.
Implements # (Bluesky auto-poster module, priority: high)

# Related PRs (if any):
This backend PR is paired with the frontend PR that adds the Bluesky auto-poster UI to the Announcements dashboard.
To test this backend PR, make sure the frontend PR [#3645](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/3645) is checked out and running.

# Main changes explained:
Create blueskyController.js: Handles login, session check, post creation, post fetching, and post deletion.
Create blueskyService.js: Uses the @atproto/api to manage Bluesky sessions and post actions.
Add blueskyRouter.js: Defines Express routes under /api/bluesky.
Support for image uploads (GIF support in progress): Uses multer with size limit of 1MB for images; GIF upload handling is under development.
Session authentication: Stores accessJwt, refreshJwt, and did in express-session per user.
Auto-refresh session: Session is refreshed automatically if older than 1 hour using the refreshJwt.
MIME and blob upload: Uploads image blobs directly using agent.uploadBlob().

# How to test:
Check into this current backend branch
Get Bluesky App Password
To authenticate with the Bluesky API, you'll need an App Password:
👉 Go to: https://bsky.app/settings/app-passwords
1. Log in with your handle (e.g. yourname.bsky.social)
2. Click "Create App Password"
3. Copy the generated password and use it for login at "Bluesky Manager" - http://localhost:3000/announcements
⚠️ This password is different from your normal login password — it's a special token used for API access.

Run backend: npm install && npm run dev

linh2020 committed:
Bluesky auto-poster

linh2020 committed:
Final commit before schedule_post feature

Anusha-Gali reviewed (changes requested):
Hi Linh,

I tried to review your PR locally and was not able to login to verify. Initially assessment was the presence of old DB.js file which i updated to the latest one yet i was unable to login. 

<img width="1278" height="824" alt="Image" src="https://github.com/user-attachments/assets/a6501822-9335-49a4-9a42-df695d7eac13" />
<img width="941" height="824" alt="Image" src="https://github.com/user-attachments/assets/4f5ecc32-3d25-4b76-8efe-0533ae47ce9d" />

linh2020 committed:
fix(login): allow Vite frontend origin and stabilize backend startup

- include localhost/127.0.0.1:5173 in backend CORS allowlist to fix /api/login from frontend

- keep credentials/auth headers and allow PATCH in CORS methods

- reduce dev startup noise by limiting babel-node transpilation to src only

- improve Mongo startup config with encoded username and deprecation-safe mongoose options

linh2020 committed:
fix(login): unblock frontend auth and resolve backend lint errors

linh2020 committed:
ci(lint): ignore node_modules and build artifacts in backend lint

linh2020 committed:
fix(lint): resolve CI parse failures and harden lint ignores

linh2020 committed:
fix: resolve SonarQube issues in Bluesky autoposter backend

linh2020 committed:
fix: resolve package-lock.json merge conflict

linh2020 committed:
Merge origin/development into Linh_Bluesky_autoposter_backend

Anusha-Gali reviewed (approved):
Hi Linh,

Thank you for updating the PR's, i have reviewed and mentioned my comments in the frontend, do have a look: https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/3645#pullrequestreview-4041750324

linh2020 committed:
Merge branch 'development' into Linh_Bluesky_autoposter_backend

linh2020 committed:
fix: remove duplicate invTypeBase destructured import in routes.js

- Removed duplicate const destructuring block for buildingInventoryType
- Duplicate 'invTypeBase' declaration caused ESLint parsing error
- Fixes Node.js CI / Lint Check failure

one-community commented:
Thank you all, merging!
